### ISSUE TITLE
Ability to run as nonroot user

### ISSUE THREAD

**Issue by johnypony3** [hashicorp/docker-consul#103]
#### Overview of the Issue

We are using consul on a kubernetes cluster. We have a [pod security policy](https://kubernetes.io/docs/concepts/policy/pod-security-policy/) that requires the pod to run under as a nonroot user. When the application is stood up, the logs contain the following error: `chown: /consul/config: Operation not permitted`. This is because `chown` is not available to the nonroot user account.

#### Reproduction Steps

1. Standup consul in kubernetes with a nonroot security context as below
```yaml
#pod
securityContext:
  fsGroup: 10001
  runAsNonRoot: true
  runAsUser: 10001

#container
securityContext:
  allowPrivilegeEscalation: false
  capabilities:
    drop:
    - ALL
  privileged: false
  runAsNonRoot: true
  runAsUser: 10001
```

1. View logs, notice error:  `chown: /consul/config: Operation not permitted`

### Operating system and Environment details

Kubernetes, Consul container

### Log Fragments

chown: /consul/config: Operation not permitted

**Comment by johnypony3** [comment]
I am able to override the entrypoint during deploy which does not perform restricted operations, but this is not ideal.

**Comment by canhnt** [comment]
The consul docker does not specify user id when creating user. From my check the `consul` user has `uid=100`, not 10001. The `consul` group gid is `1000`.

Actually it's not ideal to create user/group without ids. It makes configurations harder when in k8s, manifests rely on ids rather than names.

**Comment by loicmathieu** [comment]
Hi,
The id for user and group is a non-blocking issue.
The real issue is that docker-entrypoint.sh can only works with root user.

There is a solution found on SOF : https://stackoverflow.com/questions/50137167/run-consul-docker-image-in-kubernetes-as-non-root/50354067#50354067
It links to this github repo that have a commit that solved it : https://github.com/DatzAtWork/docker-consul/commit/bd3d2248e871d0ffa89c86edf8d7934251559df6

I can prepare a pull request, but I think the best is to contact the author of this commit and ask him to open a pull request if you will be OK to merge it.

It's a real issue for us as we plan to deploy Consul on OpenShift and cannot run our Pod as root.

Regards

**Comment by jheiss** [comment]
PR in #113

**Comment by AlexisLessard** [comment]
An ad-hoc solution may be to run `oc adm policy add-scc-to-user anyuid -z default` as a cluster administrator, but it is not advisable in production, as it allows containers to run as root, which is not secure.

**Comment by mkeeler** [comment]
Fixed in #129. (And will be pushed with the next Consul release)

To run as non-root two things will be needed:

1. The docker container needs to be run with the `CONSUL_DISABLE_USER_MGMT` set: If executing docker directly it would look like: `docker run -e CONSUL_DISABLE_PERM_MGMT= consul:latest`
2. You will need to ensure that the permissions on the data and conf directories permit the user the container is being run as to write to them. The assumption is that volumes or other bind mounts will be mounted over top of the directories and the permission management for those will be handled separately.
